travis.sh

```diff
@@ -6,7 +6,7 @@ for package in $(cat sources.txt) doc/tutorial ; do
   echo testing $package
   pushd $package
   tinc
-  cabal configure --enable-tests --disable-optimization
+  cabal configure --enable-tests --disable-optimization --ghc-options='-Wall -Werror'
```

Comment:
Good point. I've removed `-Wall` from the script and cleaned up the `ghc-options` in the individual cabal files. (I've removed all the `-fno-warn-...` flags, since they weren't necessary to get warning free builds right now. But it should be possible to add them back in without being overridden by the travis script.)